userdb/userdb.go

```diff
@@ -0,0 +1,6 @@
+package userdb
+
+type Store interface {
+	Check(username, password string) error
```

Comment:
as discussed on gitter, we'll go with what I have for now.  It won't work for the digest auth case, but it will work for database/salt/pepper.  We can revisit this interface when support for digest auth is added.